src/main/java/com/powsybl/openloadflow/util/Reports.java

```diff
@@ -44,85 +52,207 @@ public static void reportNetworkBalance(Reporter reporter, double activeGenerati
+                .withSeverity(TypedValue.ERROR_SEVERITY)
                 .build());
     }
 
-    public static void reportMismatchDistributionFailure(Reporter reporter, int iteration, double remainingMismatch) {
+    public static void reportMismatchDistributionFailure(Reporter reporter, double remainingMismatch) {
         reporter.report(Report.builder()
                 .withKey("mismatchDistributionFailure")
-                .withDefaultMessage("Iteration ${iteration}: failed to distribute slack bus active power mismatch, ${mismatch} MW remains")
-                .withValue(ITERATION, iteration)
+                .withDefaultMessage("Failed to distribute slack bus active power mismatch, ${mismatch} MW remains")
                 .withTypedValue("mismatch", remainingMismatch, OpenLoadFlowReportConstants.MISMATCH_TYPED_VALUE)
                 .withSeverity(TypedValue.ERROR_SEVERITY)
                 .build());
     }
 
-    public static void reportMismatchDistributionSuccess(Reporter reporter, int iteration, double slackBusActivePowerMismatch, int iterationCount) {
+    public static void reportMismatchDistributionSuccess(Reporter reporter, double slackBusActivePowerMismatch, int iterationCount) {
+        String plural = (iterationCount > 1) ? "s" : "";
         reporter.report(Report.builder()
                 .withKey("mismatchDistributionSuccess")
-                .withDefaultMessage("Iteration ${iteration}: slack bus active power (${initialMismatch} MW) distributed in ${iterationCount} iterations")
-                .withValue(ITERATION, iteration)
+                .withDefaultMessage("Slack bus active power (${initialMismatch} MW) distributed in ${iterationCount} inner iteration" + plural)
                 .withTypedValue("initialMismatch", slackBusActivePowerMismatch, OpenLoadFlowReportConstants.MISMATCH_TYPED_VALUE)
                 .withValue("iterationCount", iterationCount)
                 .withSeverity(TypedValue.INFO_SEVERITY)
                 .build());
     }
 
-    public static void reportNoMismatchDistribution(Reporter reporter, int iteration) {
+    public static void reportPvToPqBuses(Reporter reporter, int pvToPqBusCount, int remainingPvBusCount) {
+        String defaultMessageBeginning = pvToPqBusCount == 1 ?
+                "${pvToPqBusCount} bus switched PV -> PQ " :
+                "${pvToPqBusCount} buses switched PV -> PQ ";
+        String defaultMessageEnd = remainingPvBusCount == 1 ?
+                "(${remainingPvBusCount} bus remains PV)" :
+                "(${remainingPvBusCount} buses remain PV)";
         reporter.report(Report.builder()
-                .withKey("NoMismatchDistribution")
-                .withDefaultMessage("Iteration ${iteration}: already balanced")
-                .withValue(ITERATION, iteration)
+                .withKey("switchPvPq")
+                .withDefaultMessage(defaultMessageBeginning + defaultMessageEnd)
```

Comment:
same / don't do that, all reports with the same key MUST have the same defaultMessage